docs/_docs/03-ug-mutable-dataset.md

```diff
@@ -11,28 +11,81 @@ This guide helps you to utilize indexes on mutable dataset.
-deleted files under the root paths of the given source data.
+Since Hyperspace v0.3, we support file-level "mutable" dataset which means users can append new data files and/or
+delete existing data files under the root paths of the given source data of indexes.
 
-Previously, any change in the original dataset content, such as adding a new source data file or 
-removing an existing one, required a full refresh to make the index usable again. This could be
-a costly operation due to shuffling and repartitioning all records in the latest source data.
+Previously, any change in the original dataset content required a full refresh to make the index usable again.
+This could be a costly operation due to shuffling and repartitioning all records in the latest source data.
 
-Now, we offer several options to handle appended and deleted files more efficiently.
+Now, we offer several options to handle above scenario more efficiently.
 
 ## Options of using index when your dataset changes
 
 ### Refresh Index
-
-We provide several refresh modes to refresh index data.
+You can refresh an index according to its latest source data files by running the `refresh` command.
+Hyperspace provide several modes to refresh an index. These modes differ in terms of the way they update the index and the amount of data scans and shuffle each does.
+You should pick a mode for refreshing an index according to its current size and total amount of data deleted from or appended to its source data files.
+You can specify the mode as an argument when calling the `refresh` command.
+Currently, there are two refresh modes available for an index: `"full"` and `"incremental"`.
 
 #### Full
-TODO
+After some changes in an index's original dataset files, using `refresh` with the `"full"` mode causes
+the index refresh action perform a full rebuild of the index.
+This ends up creating a new version of the index and involves a full scan and shuffle of its latest source data.
+As a result, the amount of time it takes to refresh an index in this mode is similar to creating a new index, with the same index configuration, on the latest dataset content.
+The advantage of using the full mode is that once index refresh is finished successfully, new index files are organized in the most optimized way according to index's latest source data content and its bucketing configuration.
+
+Assume you have an index with the name `empIndex`. After adding and removing some data files from the dataset `empIndex` is created on, you can refresh it in the full mode as below:
+
+Scala:
+```scala
+import com.microsoft.hyperspace._
+
+val hs = new Hyperspace(spark)
+hs.refreshIndex("empIndex", "full")
+``` 
+
+Python:
+
+```python
+from hyperspace import Hyperspace
+
+hs = Hyperspace(spark)
+hs.refreshIndex("empIndex", "full")
+```
 
 #### Incremental
-TODO
+After some changes in an index's original dataset files, using `refresh` with the `"incremental"` mode causes
```

Comment:
"After some files are added or deleted to the original source files (from which an index was built),"?